qkernel/src/lib.rs

```diff
@@ -332,6 +330,113 @@ pub extern "C" fn syscall_handler(
     }
 }
 
+// syscall_handler implementation for aarch64:
+// Unlike x86, this function is NOT directly called 
+// from the asm code (vector). The C calling convention
+// is not necessary here.
+// TODO move this function to a proper place.
+#[no_mangle]
+#[cfg(target_arch = "aarch64")]
+pub fn syscall_dispatch_aarch64(
+    call_no: u32,
+    _arg0: u64,
+    _arg1: u64,
+    _arg2: u64,
+    _arg3: u64,
+    _arg4: u64,
+    _arg5: u64,
+) -> u64 {
+    CPULocal::Myself().SetMode(VcpuMode::Kernel);
+
+    let currTask = task::Task::Current();
+    currTask.AccountTaskLeave(SchedState::RunningApp);
+
+    let mut nr = call_no as u64;
+
+    let startTime = TSC.Rdtsc();
+    let enterAppTimestamp = CPULocal::Myself().ResetEnterAppTimestamp() as i64;
+    let worktime = Tsc::Scale(startTime - enterAppTimestamp) * 1000; // the thread has used up time slot
+    if worktime > CLOCK_TICK {
+        taskMgr::Yield();
+    }
+
+    let res;
+    let args = SyscallArguments {
+        arg0: _arg0,
+        arg1: _arg1,
+        arg2: _arg2,
+        arg3: _arg3,
+        arg4: _arg4,
+        arg5: _arg5,
+    };
+
+    let mut tid = 0;
+    let mut pid = 0;
+    let mut callId: SysCallID = SysCallID::UnknowSyscall;
+
+    let debugLevel = SHARESPACE.config.read().DebugLevel;
+
+    if debugLevel > DebugLevel::Error {
+        let llevel = SHARESPACE.config.read().LogLevel;
+        callId = if nr < SysCallID::UnknowSyscall as u64 {
+            unsafe { mem::transmute(nr as u64) }
+        } else {
+            nr = SysCallID::UnknowSyscall as _;
+            SysCallID::UnknowSyscall
+        };
+
+        if llevel == LogLevel::Complex {
+            tid = currTask.Thread().lock().id;
+            pid = currTask.Thread().ThreadGroup().ID();
+            info!("({}/{})------get call id {:?} arg0:{:x}, 1:{:x}, 2:{:x}, 3:{:x}, 4:{:x}, 5:{:x}, userstack:{:x}, return address:{:x}, fs:{:x}",
+                tid, pid, callId, _arg0, _arg1, _arg2, _arg3, _arg4, _arg5, currTask.GetPtRegs().rsp, currTask.GetPtRegs().rcx, GetFs());
```

Comment:
I think it may not compile if we define a PtRegs in aarch64, and I think maybe we have to union the definition of `ExceptionFrame` and `PtRegs`, because qkernel also get this PtRegs from the stack.  I made a temporary definition of `PtRegs` in aarch64 in this PR https://github.com/QuarkContainer/Quark/pull/1027, could you please take a look?

Reply:
regarding the `GetPtRegs`, I'm aware that this won't work for aarch64 right now. For now the `syscall_dispatch_aarch64` is a mere copy of the x86 syscall handler code (i.e. placeholder).

I'll get this fixed when we pin down the `ExceptionFrame` definitions. 

> could you please take a look?

Yes, and thanks for the ping. Cheers.